Mathlib/AlgebraicGeometry/PrimeSpectrum/ConstructibleSet.lean

```diff
@@ -0,0 +1,80 @@
+/-
+Copyright (c) 2024 Yaël Dillies, Andrew Yang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Andrew Yang
+-/
+import Mathlib.AlgebraicGeometry.PrimeSpectrum.Basic
+import Mathlib.Order.SuccPred.WithBot
+
+/-!
+# Constructible sets in the prime spectrum
+
+This file provides tooling for manipulating constructible sets in the prime spectrum of a ring.
+
+## TODO
+
+Link to constructible sets in a topological space.
+-/
+
+open Finset
+open scoped Polynomial
+
+namespace PrimeSpectrum
+variable {R S T : Type*} [CommRing R] [CommRing S] [CommRing T]
+
+variable (R) in
+/-- The data of a constructible set `s` is finitely many tuples `(f, g₁, ..., gₙ)` such that
+`s = ⋃ (f, g₁, ..., gₙ), V(g₁, ..., gₙ) \ V(f)`.
+
+To obtain `s` from its data, use `PrimeSpectrum.ConstructibleSetData.toSet`. -/
+abbrev ConstructibleSetData := Finset (Σ n, R × (Fin n → R))
+
+namespace ConstructibleSetData
+
+/-- Given the data of the constructible set `s`, build the data of the constructible set
+`{I | {x | f x ∈ I} ∈ s}`. -/
+def map [DecidableEq S] (f : R →+* S) (s : ConstructibleSetData R) : ConstructibleSetData S :=
+  s.image <| Sigma.map id fun _n (r, g) ↦ (f r, f ∘ g)
+
+@[simp] lemma map_id [DecidableEq R] (s : ConstructibleSetData R) : s.map (.id _) = s := by
+  unfold map Sigma.map; simp [RingHom.coe_id]
+
+lemma map_comp [DecidableEq S] [DecidableEq T] (f : S →+* T) (g : R →+* S)
+    (s : ConstructibleSetData R) : s.map (f.comp g) = (s.map g).map f := by
+  unfold map Sigma.map; simp [image_image, Function.comp_def]
+
+open scoped Classical in
+/-- Remove the zero polynomials from the data of a constructible set. -/
+noncomputable def reduce (S : ConstructibleSetData R) : ConstructibleSetData R :=
+  S.image fun ⟨_, r, f⟩ ↦ ⟨#{x | f x ≠ 0}, r, f ∘ Subtype.val ∘ (Finset.equivFin _).symm⟩
```

Comment:
Do we still need this?

Reply:
No, you're right. I had completely overlooked that it wasn't used anymore